android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/BootReceiver.kt

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.firefox.vpn.daemon
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+
+class BootReceiver : BroadcastReceiver() {
+    private val TAG = "BootReceiver"
+
+    override fun onReceive(context: Context, arg1: Intent) {
+        Log.init(context)
+        if (!Prefs.get(context).getBoolean(START_ON_BOOT, false)) {
+            Log.i(TAG, "This device did not enable start on boot - exit")
+            return
+        }
+        Log.i(TAG, "This device did enable start on boot - try to start")
+        val intent = Intent(context, VPNService::class.java)
+        intent.putExtra("startOnBoot", true)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
Just to see if I get it. We are checking if the current Android SDK version is greater than Android SDK version `Oreo`? Ha... That took me a while, why does 0 look so much like O???????? Anyways, would you mind adding a comment here on why we need to check this?